src/libraries/System.CodeDom/tests/System/CodeDom/Compiler/VBCodeGenerationTests.cs

```diff
@@ -3264,5 +3264,63 @@ End Sub
                       End Class
                   End Namespace");
         }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "The bug was present on .NET Framework: https://github.com/dotnet/runtime/issues/56267")]
+        public void OrdinaryCommentsDoNotAccidentallyBecomeDocumentationComments()
+        {
+            var codeTypeDeclaration = new CodeTypeDeclaration("ClassWithCommment")
+            {
+                IsClass = true,
+                Comments =
+                {
+                    new CodeCommentStatement(
+                        "'' Lines starting with exactly two single quotes" + Environment.NewLine +
+                        "'' each get a separating space," + Environment.NewLine +
+                        "but other lines do not get a space. This way generated files only change on tool upgrade where there were generation bugs." + Environment.NewLine +
+                        "' Not even lines starting with only one single quote" + Environment.NewLine +
+                        "''' or three single quotes.",
+                        docComment: false),
+                },
+            };
+
+            AssertEqualPreserveLineBreaks(codeTypeDeclaration,
+                @"
+                  ' '' Lines starting with exactly two single quotes
+                  ' '' each get a separating space,
+                  'but other lines do not get a space. This way generated files only change on tool upgrade where there were generation bugs.
+                  '' Not even lines starting with only one single quote
+                  '''' or three single quotes.
+                  Public Class ClassWithCommment
+                  End Class
+                ");
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "The bug was present on .NET Framework: https://github.com/dotnet/runtime/issues/56267")]
+        public void DocumentationCommentsDoNotAccidentallyBecomeOrdinaryComments()
+        {
+            var codeTypeDeclaration = new CodeTypeDeclaration("ClassWithCommment")
+            {
+                IsClass = true,
+                Comments =
+                {
+                    new CodeCommentStatement(
+                        "' Lines starting with a single quote" + Environment.NewLine +
+                        "' each get a separating space," + Environment.NewLine +
+                        "but other lines do not get a space. This way generated files only change on tool upgrade where there were generation bugs.",
```

Comment:
I just noticed that this is invalid
```
''' <summary>
'''''starts with two quotes
''' </summary>
```
Ie a line starting with any number of quotes needs a space. I think your code does this, but we should include it in the test.

Reply:
done